packages/tasks/src/tasks/chat-completion/spec/input.json

```
@@ -0,0 +1,54 @@
+{
+	"title": "ChatCompletionInput",
+	"description": "Inputs for ChatCompletion inference",
+	"type": "object",
+	"properties": {
+		"messages": {
+			"type": "array",
+			"title": "ChatCompletionInputMessage",
+			"items": {
+				"type": "object",
+				"properties": {
+					"role": {
+						"oneOf": [{ "const": "assistant" }, { "const": "system" }, { "const": "user" }],
```

(nit) Maybe add a title here too to differentiate the type?

```suggestion
						"oneOf": [{ "const": "assistant" }, { "const": "system" }, { "const": "user" }],
						"title": "ChatCompletionInputMessageRole",
```

---

Good idea yes